### DISCUSSION TITLE
jira issue create with custom_field.

### DISCUSSION THREAD

**Discussion by abbi1229** [ankitpokhrel/jira-cli#222] [Ideas] [answered]
Not only "Priority", "Components" and "Labels", but we also have some custom field to mange the ticket.
We need to edit the field by web browser after we create a ticket.

It would be helpful, if we can create or edit the custom field by jira-cli.
If there are too many Flags, it’s also great to use config.yml (--config) to implement.

**Comment by wonderg** [comment]
I can also say that is some cases we use customfield_ as a mandatory field for tickets which actually mean that currently that is not possible to generate a ticket using **jira issue create** cmd.

I would suggest to use the similar syntax which is already implemented 
jira issue create -tBug -s"New Bug" -yHigh -lbug -lurgent -b"Bug description" -customfield_12345"Some text"

**Reply by realtime-neil** [reply]
Adding a command-line option for every custom field is probably infeasible. Adding something like a general `--set-key-value=MyKey=MyValue` might be easier. With that implemented, one could re-implement

```
-t,--type
-P,--parent
-s,--summary
-b,--body
-y,--priority
-a,--assignee
-l,--label
-C,--component
```

as aliases to

```
--set-key-value=type=my_type
--set-key-value=parent=my_parent
--set-key-value=summary=my_summary
--set-key-value=body=my_body
--set-key-value=priority=my_priority
--set-key-value=assignee=my_assignee
--set-key-value=label=my_label
--set-key-value=component=my_component
```

**Reply by ankitpokhrel** [reply]
Thank you all for the input. I would agree with the approach suggested by @realtime-neil, I know that some CLI tools (eg: [git](https://git-scm.com/docs/git#Documentation/git.txt---config-envltnamegtltenvvargt)) use a similar approach. One problem I see, however, is that we need different types of data as a value (example below):

- Checkbox custom field
  ```json
  "customfield_11440" : [{ "value" : "option1"}, {"value" : "option2"}]
  ```

- Date time picker custom field: A custom field that is a date time in ISO 8601 `YYYY-MM-DDThh:mm:ss.sTZD` format.
  ```json
  "customfield_11442" : "2015-11-18T14:39:00.000+1100"
  ```

- Cascading select custom field
  ```json
  "customfield_11447" : { "value": "parent_option1", "child": { "value" : "p1_child1"} }
  ```

- Multi-user picker custom field
  ```json
  "customfield_11458" : [ { "name":"inigomontoya" }, { "name":"tommytomtomahawk" }]
  ```

- Text custom field
  ```json
  "customfield_11450": "An example of a single line of text"
  ```

We can expect the end-user to pass data in the exact format needed for the field (eg: below), but that's not user friendly IMO, and I am pretty sure this will be a source of lots of complaints in the future.

```sh
--set-key-value=customfield_11458='[ { "name":"inigomontoya" }, { "name":"tommytomtomahawk" }]'
```

Thoughts?

**Comment by ankitpokhrel** [comment]
I worked on a proof of concept for this feature: https://github.com/ankitpokhrel/jira-cli/pull/319

The details are outlined in the PR description. It would be great if you can test this out and report any issues.

**Comment by ankitpokhrel** [comment, marked as answer]
This change is available in the new release [v1.0.0-beta](https://github.com/ankitpokhrel/jira-cli/releases/tag/v1.0.0-beta)
